### ISSUE TITLE
Network stability and blocking pop-up dialog

### ISSUE THREAD

**Issue by vadimkantorov** [xwiki-labs/cryptpad#964]
### Contribution guidelines

- [X] I've read the [code of conduct](https://github.com/xwiki-labs/cryptpad/blob/main/CODE_OF_CONDUCT.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] I understand that not following the below instructions will result in immediate closure and/or deletion of my issue.
- [X] I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [X] I have understood that answers are voluntary and community-driven, and not commercial support.
- [X] I have verified that my issue has not been already answered in the past. I've read the [Common issues documentation section](https://docs.cryptpad.fr/en/FAQ.html#common-issues) and I also checked [previous issues](https://github.com/xwiki-labs/cryptpad/issues).

### Description

It's my first time interacting with cryptpad hosted at cryptpad.fr. It's a document someone sent me.

My internet connection is fine, but cryptpad seems to lose connection every ~1 second. So it keeps changing status between Disconnected and Connected. Also, a blocking pop-up error box is appearing and disappearing every second. This is extremely annoying and renders work impossible:

![image](https://user-images.githubusercontent.com/1041752/188228078-95ca95fc-7a82-4b04-839a-1889129dd3cd.png)

### Steps to reproduce

Don't have any special steps, just opened the document and tried to read it (not even edit it)

### Expected behavior

No blocking pop-up box is appearing (even if connection seems down)

### System information

N/A

**Comment by mthld** [comment]
Your **System informations** part of the template is missing.

**Comment by vadimkantorov** [comment]
Well, I have no particular system informations, using the latest updated Chrome on Windows and cryptpad.fr website

**Comment by vadimkantorov** [comment]
After a page reload, the problem fixed itself. But it's a very unpleasant UX nonetheless

**Comment by mthld** [comment]
We ask for system informations because it matters, if you had read [the documentation page](https://docs.cryptpad.fr/en/FAQ.html#common-issues) linked in the header of the issue template you'd have seen that many "privacy" web browser extensions are interfering with CryptPad.

The problem you describe matches this kind of issue. That's why to help you we need the proper information, otherwise there is nothing we can do on our end. We can't guess what your setup is and trace an error that happened on client side without knowing nothing about.

**Comment by vadimkantorov** [comment]
Gotcha. I have "Session Buddy", "Backspace to go Back", "OneTab", "The Marvelous Suspender". 

I never noticed any other problems with them on any other website, so while they can theoretically interfere, I'd be very surprised if they do.

I assume, there're two aspects of the problem:
1) CryptPad being over-zealous in declaring "Disconnected" state after a very small timeout, and very often
2) Showing/hiding an overlay box when it happens. Even if it's disconnected, probably a read-only mode with just a status Disconnected somewhere would suffice. At least it would not interfere with reading the doc.

If this Disconnected status spamming can happen without extensions, showing the overlay box is problematic as well

**Comment by mthld** [comment]
Thanks for taking the time to break things up!

If I understand correctly, you would like us to rework the disconnection system to make it less intrusive? There is surely something we can do on that UI/UX matter.

Regarding the disconnection timeout, I believe we must check very regularly for a stable connection between clients and servers. Otherwise instant online collaboration wouldn't be possible the way we implement it, so I'm not sure about raising timeouts.

**Comment by vadimkantorov** [comment]
Yes, I think the blocking-everything-overlay-box is excessive (and not very useful) in this super-spamming Disconnect scenario, but maybe in general as well. Probably a noticeable (marked red or yellow to be noticeable) "Disconnected" somewhere would be sufficient (+optionally read-only mode if edits are problematic; e.g. I was having trouble to even click on FIle->Print to make a PDF, I managed at the end but it involved some acrobatics)

And yes, more lenient timeouts would be good (at least for turning from Disconnected back to Connected state, it .e.g should be connected for 5 seconds, otherwise it always stays Disconnected). For me for these micro-disconnects, both always-connected and always-disconnected are better than Disconnected spams

**Comment by mthld** [comment]
Happened to me while browsing a document, from Tor browser. Here is a screen recording for posterity:

https://user-images.githubusercontent.com/41020854/188643222-fa7f64c9-dc51-4ede-ac9b-f49f457287ec.mp4

**Comment by petrenkonikita112263** [comment]
Back with some update information about this issue, I don't know if the console log will be helpful.

The error is pop-up and still blinking
No error is pop-up
```
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
sframe-boot.js?ver=1.11:42 Testing if CSP correctly blocks an 'eval' call
LessLoader.js?ver=5.2.2-0:195 Compiling [/customize/src/less2/include/loading.less] took 11ms
LessLoader.js?ver=5.2.2-0:195 Compiling [/kanban/app-kanban.less] took 49ms
cryptpad-common.js?ver=5.2.2-0:2698 Outer ready
cryptpad-common.js?ver=5.2.2-0:2721 Posting CONNECT
/kanban/#/3/kanban/edit/0522fd3edbb520805253c15350a8ac3a/:1 
        
       Failed to fetch a worker script.
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
sframe-boot.js?ver=1.11:42 Testing if CSP correctly blocks an 'eval' call
LessLoader.js?ver=5.2.2-0:195 Compiling [/customize/src/less2/include/loading.less] took 8ms
LessLoader.js?ver=5.2.2-0:195 Compiling [/secureiframe/app-secure.less] took 29ms
```

```
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
​ Testing if CSP correctly blocks an 'eval' call
LessLoader.js?ver=5.2.2-0:195 Compiling [/customize/src/less2/include/loading.less] took 4ms
LessLoader.js?ver=5.2.2-0:195 Compiling [/kanban/app-kanban.less] took 10ms
cryptpad-common.js?ver=5.2.2-0:2698 Outer ready
cryptpad-common.js?ver=5.2.2-0:2721 Posting CONNECT
/kanban/#/2/kanban/edit/Satp8W+0ZxMzTVZJITB4AfhL/:1 
        
       Failed to fetch a worker script.
toolbar.js?ver=5.2.2-0:966 
        
       Unable to get the pinned usage RPC_NOT_READY
todo @ toolbar.js?ver=5.2.2-0:966
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
​ Testing if CSP correctly blocks an 'eval' call
LessLoader.js?ver=5.2.2-0:195 Compiling [/customize/src/less2/include/loading.less] took 2ms
LessLoader.js?ver=5.2.2-0:195 Compiling [/secureiframe/app-secure.less] took 5ms
jkanban_cp.js?ver=5.2.2-0:396 In drop
jkanban_cp.js?ver=5.2.2-0:362 In dragend
```
